tools/sgcargo/tools.go

```diff
@@ -0,0 +1,55 @@
+package sgcargo
+
+import (
+	"context"
+	"os"
+	"os/exec"
+
+	"go.einride.tech/sage/sg"
+	"go.einride.tech/sage/tools/sgrustup"
+)
+
+const (
+	name = "cargo"
+)
+
+//nolint:gochecknoglobals
+var Home = sg.FromToolsDir(name)
+
+func Command(ctx context.Context, args ...string) *exec.Cmd {
+	sg.Deps(ctx, PrepareCommand)
+	path := os.Getenv("PATH")
+	ctx = sg.ContextWithEnv(
+		ctx,
+		"PATH="+sg.FromToolsDir(name, "bin")+":"+path,
+		"RUSTUP_HOME="+sgrustup.Home,
+	)
+	return sg.Command(ctx, sg.FromBinDir(name), args...)
+}
+
+func PrepareCommand(ctx context.Context) error {
+	sg.Deps(ctx, sgrustup.PrepareCommand)
+	return nil
+}
+
+func FormatCheck(ctx context.Context) error {
```

Comment:
these functions are all one-liners and I would propose to just have them inlined and reduce the public API footprint here